server/src/main/java/umm3601/digitalDisplayGarden/PlantController.java

```diff
@@ -57,6 +75,87 @@ public String listPlants(Map<String, String[]> queryParams) {
         return JSON.serialize(matchingPlants);
     }
 
+    /**
+     * Takes a String representing a hexadecimal ID number of a plant
+     * and when the ID is found in the database returns a JSON document
+     * as a String of the following form
+     *
+     * <code>
+     * {
+     *  "_id"        : { "$oid": String },
+     *  "commonName" : String,
+     *  "cultivar"   : String
+     * }
+     * </code>
+     *
+     * If the ID is invalid or not found, the following JSON value is
+     * returned
+     *
+     * <code>
+     *  null
+     * </code>
+     *
+     * @param id a hexadecimal ID number of a plant in the DB
+     * @return a string representation of a JSON value
+     */
+    public String getPlant(String id) {
+
+        FindIterable<Document> jsonPlant;
+        String returnVal;
+        try {
+
+            jsonPlant = plantCollection.find(eq("_id", new ObjectId(id)))
+                    .projection(fields(include("commonName", "cultivar")));
+
+            Iterator<Document> iterator = jsonPlant.iterator();
+
+            if (iterator.hasNext()) {
+                //incrementMetadata(id, "pageViews");
+                returnVal = iterator.next().toJson();
+            } else {
+                returnVal = "null";
+            }
+
+        } catch (IllegalArgumentException e) {
+            returnVal = "null";
+        }
+
+        return returnVal;
+
+    }
+
+
+    /**
+     * Finds a plant and atomically increments the specified field
+     * in its metadata object. This method returns true if the plant was
+     * found successfully (false otherwise), but there is no indication of
+     * whether the field was found.
+     *
+     * @param id a hexadecimal ID number of a plant in the DB
+     * @param field a field to be incremented in the metadata object of the plant
+     * @return true if a plant was found
+     * @throws com.mongodb.MongoCommandException when the id is valid and the field is empty
+     */
+    public boolean incrementMetadata(String id, String field) {
```

Comment:
Let's wait on borrowing the `incrementMetadata` method since we are working on ratings in a different branch.
See https://github.com/UMM-CSci-3601-S17/digital-display-garden-iteration-2-grimaldi/commit/adf32373b98c19344d420c2b0270c1801d220446#diff-cfaf55f4e16589d95ffb5f5ad1b61692